src/EmbedIndex/Program.cs

```diff
@@ -164,33 +165,85 @@ public static IEnumerable<string> ComputeIndexEntries(string archiveRelativePath
             return keys;
         }
 
+        private const string SymbolCatalogFileName = "signatures.cat";
+        private const string CatalogedSymbolListFileName = "cataloged.txt";
+        private const string CatalogIndexFileName = "catalog_index_id.txt";
+
         private static IEnumerable<Tuple<string, string>> ComputeCatalogEntries(ZipArchive archive, IEnumerable<IFileFormatIndexer> indexers)
         {
-            foreach (ZipArchiveEntry catalogEntry in archive.Entries)
+            // create catalog_index_id.txt with SHA-256 of catalog
+            // store in root
+            ZipArchiveEntry catalogEntry = archive.GetEntry(SymbolCatalogFileName);
+            
+            if (catalogEntry == null)
+            {
+                yield break;
+            }
+
+            string catalogHash;
+            using (Stream catalogStream = catalogEntry.Open())
             {
-                if (!catalogEntry.Name.ToLowerInvariant().EndsWith(".cat"))
+                using (HashAlgorithm sha = SHA256.Create())
                 {
-                    continue;
+                    catalogHash = sha.ComputeHash(catalogStream).ToHexString();
+                    ZipArchiveEntry catalogIndexEntry = archive.CreateEntry(CatalogIndexFileName);
+                    using (Stream indexStream = catalogIndexEntry.Open())
+                    {
+                        using (TextWriter writer = new StreamWriter(indexStream))
+                        {
+                            writer.WriteLine(catalogHash);
+                            writer.Flush();
+                        }
+                    }
                 }
-                string signedFile = catalogEntry.FullName.Substring(0, catalogEntry.FullName.Length - 4);
-                ZipArchiveEntry signedFileEntry = archive.Entries.SingleOrDefault(e => e.FullName == signedFile);
-                if (signedFileEntry == null)
+            }
+
+            // get list of cataloged symbol files
+            ZipArchiveEntry catalogedFilesEntry = archive.GetEntry(CatalogedSymbolListFileName);
+
+            if (catalogedFilesEntry == null)
```

Comment:
Do we want to embed the `catalog_index_id.txt` even if `catalogedFilesEntry == null`? Might want to move this check to the beginning of the method.